### PR TITLE
#48: Add MyPy to first CI

### PR DESCRIPTION
**Pull request by jakejack13**
Implements: #48

### CODE REVIEW

File: microservices/first/main.py [hunk 1]
```diff
@@ -1,27 +1,28 @@
 # Endpoint documentation: https://github.com/jakejack13/SPEED/blob/microservices/first_api_doc.md
-import json
 from utils import DBManager
 import utils
 
 from flask import Flask, request, jsonify
+import flask
```
Comment (owenxr) [inline microservices/first/main.py:6]: double import of flask?

**Review by owenxr — CHANGES_REQUESTED** [review]

File: microservices/first/main.py (hunk 1 above)
Comment (jakejack13) [reply microservices/first/main.py:6]: This way, I can use `flask.Response` in a type annotation without MyPy complaining that I haven't used `Response`

File: microservices/first/main.py [hunk 2]
```diff
@@ -1,27 +1,28 @@
 # Endpoint documentation: https://github.com/jakejack13/SPEED/blob/microservices/first_api_doc.md
-import json
 from utils import DBManager
 import utils
 
 from flask import Flask, request, jsonify
```
Comment (owenxr) [inline microservices/first/main.py:5]:
```suggestion
from flask import Flask, request, jsonify, Response
```
Shouldn't it be this instead then?

**Review by owenxr — CHANGES_REQUESTED** [review]

File: microservices/first/main.py (hunk 2 above)
Comment (owenxr) [reply microservices/first/main.py:5]: Then the type annotation is just `Response`

File: microservices/first/main.py [hunk 3]
```diff
@@ -1,27 +1,28 @@
...
-@app.before_first_request
-def initialize():
+def initialize() -> None:
     """Initialize the database and tables."""
     db.create_deployments_table()
     db.create_results_table()
 
 @app.route('/start', methods=['POST'])
-def start_deployment():
+def start_deployment() -> tuple[flask.Response, int]:
```
Comment (owenxr) [inline microservices/first/main.py]:
```suggestion
def start_deployment() -> tuple[Response, int]:
```

File: microservices/first/main.py [hunk 4]
```diff
@@ -30,7 +31,7 @@ def start_deployment():
     return jsonify({"id": leader_id}), 201
 
 @app.route('/info/<int:leader_ID>', methods=['GET'])
-def get_deployment_info(leader_ID):
+def get_deployment_info(leader_ID: int) -> tuple[flask.Response, int]:
```
Comment (owenxr) [inline microservices/first/main.py]:
```suggestion
def get_deployment_info(leader_ID: int) -> tuple[Response, int]:
```

File: microservices/first/main.py [hunk 5]
```diff
@@ -50,15 +51,18 @@ def get_deployment_info(leader_ID):
         return jsonify({"error": "Deployment not found"}), 404
 
 @app.route('/update/<int:leader_ID>', methods=['POST'])
-def update_deployment(leader_ID):
+def update_deployment(leader_ID: int) -> tuple[flask.Response, int]:
```
Comment (owenxr) [inline microservices/first/main.py]:
```suggestion
def update_deployment(leader_ID: int) -> tuple[Response, int]:
```

File: microservices/first/main.py [hunk 6]
```diff
@@ -50,15 +51,18 @@ def get_deployment_info(leader_ID):
...
-    db.update_deployment_fields(leader_ID, json.loads(request.json))
+    data = request.json
+    if data is None:
+        return jsonify({'error': 'Missing request body'}), 400
+    db.update_deployment_fields(leader_ID, data)
     return jsonify({"message": "Deployment updated successfully"}), 200
 
 @app.route('/add_results/<int:leader_ID>', methods=['POST'])
-def add_results(leader_ID):
+def add_results(leader_ID: int) -> tuple[flask.Response, int]:
```
Comment (owenxr) [inline microservices/first/main.py]:
```suggestion
def add_results(leader_ID: int) -> tuple[Response, int]:
```

**Review by owenxr — APPROVED** [review]
LGTM

**Pull request merged**